File: flight_create_orders/v1/post/python/flight_create_orders.py
# Install the Python library from https://pypi.org/project/amadeus
from amadeus import Client, ResponseError

amadeus = Client(
    client_id='YOUR_AMADEUS_API_KEY',
    client_secret='YOUR_AMADEUS_API_SECRET'
)

traveler = {
  'id': '1',
  'dateOfBirth': '1982-01-16',
  'name': {
    'firstName': 'JORGE',
    'lastName': 'GONZALES'
  },
  'gender': 'MALE',
  'contact': {
    'emailAddress': 'jorge.gonzales833@telefonica.es',
    'phones': [{
      'deviceType': 'MOBILE',
      'countryCallingCode': '34',
      'number': '480080076'
    }]
  },
  'documents': [{
    'documentType': 'PASSPORT',
    'birthPlace': 'Madrid',
    'issuanceLocation': 'Madrid',
    'issuanceDate': '2015-04-14',
    'number': '00000000',
    'expiryDate': '2025-04-14',
    'issuanceCountry': 'ES',
    'validityCountry': 'ES',
    'nationality': 'ES',
    'holder': True
  }]
}

try:
    # Flight Offers Search to search for flights from MAD to ATH
    flight_search = amadeus.shopping.flight_offers_search.get(originLocationCode='MAD',
                                                              destinationLocationCode='ATH',
                                                              departureDate='2020-12-01',
                                                              adults=1).data

    # Flight Offers Price to confirm the price of the chosen flight
    price_confirm = amadeus.shopping.flight_offers.pricing.post(flight_search[0]).data

    # Flight Create Orders to book the flight
    booked_flight = amadeus.booking.flight_orders.post(flight_search[0], traveler).data

except ResponseError as error:
    raise error

File: flight_create_orders/v1/post/Node SDK/flight_create_orders.js
var Amadeus = require('amadeus');
var amadeus = new Amadeus({
  clientId: 'YOUR_API_KEY',
  clientSecret: 'YOUR_API_SECRET'
});

// Book a flight from MAD to ATH on 2022-08-01
amadeus.shopping.flightOffersSearch.get({
  originLocationCode: 'MAD',
  destinationLocationCode: 'ATH',
  departureDate: '2022-08-01',
  adults: '1'
}).then(function (flightOffersResponse) {
  return amadeus.shopping.flightOffers.pricing.post(
    JSON.stringify({
      "data": {
        "type": "flight-offers-pricing",
        "flightOffers": [
          flightOffersResponse.data[0]
        ]
      }
    })
  );
}).then(function (pricingResponse) {
  return amadeus.booking.flightOrders.post(
    JSON.stringify({
      'data': {
        'type': 'flight-order',
        'flightOffers': [pricingResponse.data.flightOffers[0]],
        'travelers': [{
          "id": "1",
          "dateOfBirth": "1982-01-16",
          "name": {
            "firstName": "JORGE",
            "lastName": "GONZALES"
          },
          "gender": "MALE",
          "contact": {
            "emailAddress": "jorge.gonzales833@telefonica.es",
            "phones": [{
              "deviceType": "MOBILE",
              "countryCallingCode": "34",
              "number": "480080076"
            }]
          },
          "documents": [{
            "documentType": "PASSPORT",
            "birthPlace": "Madrid",
            "issuanceLocation": "Madrid",
            "issuanceDate": "2015-04-14",
            "number": "00000000",
            "expiryDate": "2025-04-14",
            "issuanceCountry": "ES",
            "validityCountry": "ES",
            "nationality": "ES",
            "holder": true
          }]
        }]
      }
    })
  );
}).then(function (response) {
  console.log(response);
}).catch(function (response) {
  console.error(response);
});
